packages/core/src/models/MCP.ts

```diff
@@ -0,0 +1,244 @@
+   *       authConfigId: "ac_sdhkjfhjksdk",
+   *       allowedTools: ["GMAIL_FETCH_EMAILS"],
+   *     },
+   *   ],
+   *   { useManagedAuthByComposio: true },
+   * );
+   * 
+   * // Get server instance
+   * const server = await cfg.getServer({
+   *   connected_account_ids: {
+   *     "gmail": "acc_123",
+   *     "supabase": "acc_456"
+   *   }
+   * });
+   * // OR
+   * const server = await cfg.getServer({
+   *   user_id: "user_123"
+   * });
+   * ```
+   */
+  async create(
+    name: string,
+    toolkitConfigs: MCPToolkitConfig[],
+    authOptions?: MCPAuthOptions
+  ): Promise<MCPCreateMethodResponse> {
+    // Check for unique toolkits
+    const toolkits = toolkitConfigs.map(config => config.toolkit);
+    const uniqueToolkits = new Set(toolkits);
+    if (uniqueToolkits.size !== toolkits.length) {
+      throw new Error('Duplicate toolkits are not allowed. Each toolkit must be unique.');
+    }
+
+    // Create a custom MCP server with the toolkit configurations
+    const mcpServerCreatedResponse: CustomCreateResponse = await this.client.mcp.custom.create({
+      name,
+      toolkits: toolkits,
+      custom_tools: toolkitConfigs.flatMap(config => config.allowedTools),
+      managed_auth_via_composio: authOptions?.useManagedAuthByComposio || false,
+      auth_config_ids: toolkitConfigs.map(config => config.authConfigId)
+    });
+
+    // Add getServer method to response
+    return {
+      ...mcpServerCreatedResponse,
+      toolkits,
+      getServer: async (params: MCPGetServerParams): Promise<GenerateURLResponse> => {
+        // Validate that only one of user_id or connected_account_ids is provided
+        if (params.user_id && params.connected_account_ids) {
+          throw new Error('Cannot specify both user_id and connected_account_ids. Please provide only one.');
+        }
+
+        if (!params.user_id && !params.connected_account_ids) {
+          throw new Error('Must provide either user_id or connected_account_ids.');
+        }
+
+        // If connected_account_ids is provided, verify toolkit names
+        if (params.connected_account_ids) {
+          const providedToolkits = Object.keys(params.connected_account_ids);
+          const invalidToolkits = providedToolkits.filter(toolkit => !toolkits.includes(toolkit));
+          if (invalidToolkits.length > 0) {
+            throw new Error(`Invalid toolkits provided: ${invalidToolkits.join(', ')}. Available toolkits are: ${toolkits.join(', ')}`);
+          }
+        }
+
+        const generateURLParams = {
+          mcp_server_id: mcpServerCreatedResponse.id,
+          user_ids: params.user_id ? [params.user_id] : [],
+          connected_account_ids: params.connected_account_ids ? Object.values(params.connected_account_ids) : [],
+          managed_auth_by_composio: authOptions?.useManagedAuthByComposio || false
+        }
+        return this.client.mcp.generate.url(generateURLParams);
+      }
+    } as MCPCreateMethodResponse;
```

Comment:
Using `return {} as Type` here bypasses type safety and can lead to runtime errors if required fields are missing. It gives a false sense of correctness and may cause issues during refactoring. Remove `as type` and consider constructing a valid object or use zod.parse().

Reply:
fixed it